upgrade-plugins/src/main/java/org/exoplatform/ecms/upgrade/templates/NodeTypeTemplateUpgradePlugin.java

```diff
@@ -91,11 +120,31 @@ public void processUpgrade(String oldVersion, String newVersion) {
       // remove all old node type nodes
       for (Node removedNode : removedNodes) {
         try {
-          removedNode.remove();
+          String removedTemplateName = removedNode.getName();
+          //if Template had not been edited before, remove it
+          if(!modifiedTemplateLog.contains(removedTemplateName)){
+            if (log.isInfoEnabled()) {
+              log.info("old template {} will be removed, and new template will be imported", removedTemplateName);
+            }
+            removedNode.remove();
+          }else{
+            //else if Template was edited before, rename it
+            if (log.isWarnEnabled()) {
+              log.warn("Views, Dialogs, Skins of old template {} will be renamed, and new ones will be imported",
```

Comment:
"Templates of {} have been customized. They will be updated by the new version included in eXo Platform 4.3 but your customized templates will be kept and renamed. If you want to re-apply your customizations to the new templates versions, you can retrieve them in the Content Administration."